third_party/proton/proton/viewer.py

```diff
@@ -133,6 +133,11 @@ def get_time_seconds(df):
             gf.dataframe[f"{metric} (inc)"] = (get_time_seconds(gf.dataframe) /
                                                time_factor_dict.factor[metric_time_unit])
             derived_metrics.append(f"{metric} (inc)")
+
+            kernel_time = get_time_seconds(gf.dataframe)
```

Comment:
I think the percentage/ratio metric is a bit different from other metrics. Essentially, you can calculate a percentage/ratio for every kind of metric.

So first I would suggest using a `%` sign to represent percentage so that a metric name will become `<metric-name>/%` which can be uniformly applied to every kind of metric.

Next we check if the metric contains `/%` or not, if so, we apply the following transformation:
```
total = gf.dataframe[metric_name].iloc[0]
gf.dataframe[f"{metric}/% (inc)"] = gf.dataframe[metric_name] / total * 100.0
```

What do you think?

Reply:
In general I think that makes but what would something like tflop/s percentage of the whole model tell you? Is that even a valid metric? Take the matmul Proton tutorial for example you get something like this

```
0.649 ROOT
├─ nan _ZN2at6native12_GLOBAL__N_143distribution_elementwise_grid_stride_kernelIfLi4EZNS0_9templates4cuda20normal_and_transformIN3c104HalfEfLm4EPNS_17CUDAGeneratorImplEZZZNS4_13normal_kernelIS9_EEvRKNS_10TensorBaseEddT_ENKUlvE_clEvENKUlvE1_clEvEUlfE_EEvRNS_18TensorIteratorBaseET2_T3_EUlP25hiprandStatePhilox4_32_10E0_ZNS1_27distribution_nullary_kernelIS7_fLi4ES9_SO_SH_EEvSJ_SK_RKSL_T4_EUlifE_EEviNS_15PhiloxCudaStateET1_SK_
├─ 1.264 matmul_1024_1024_1024
│  ├─ nan _ZN2at6native29vectorized_elementwise_kernelILi4ENS0_11FillFunctorIiEENS_6detail5ArrayIPcLi1EEEEEviT0_T1_
│  ├─ 86.745 cublas
│  │  └─ nan Cijk_Ailk_Bljk_HHS_BH_MT128x96x64_MI32x32x8x1_SE_1LDSB0_APM1_ABV0_ACED0_AF0EM8_AF1EM1_AMAS0_ASE_ASGT_ASLT_ASM_ASAE01_ASCE01_ASEM1_AAC0_BL1_BS1_CLR0_DTLA0_DTLB0_DTVA0_DTVB0_DVO0_ETSP_EPS0_ELFLR0_EMLL0_FSSC10_FL0_GLVWA8_GLVWB8_GRCGA1_GRCGB1_GRPM1_GRVW8_GSU1_GSUASB_GLS0_ISA90a_IU1_K1_KLA_LBSPPA0_LBSPPB128_LPA0_LPB8_LDL1_LRVW8_LWPMn1_LDW0_MAC_MIAV0_MDA2_MO40_MMFGLC_MKFGSU256_NTA0_NTB0_NTC0_NTD0_NEPBS0_NLCA1_NLCB1_ONLL1_OPLV0_PK0_PAP0_PGR2_PLR9_PKA0_SIA3_SLW1_SS0_SU0_SUM0_SUS0_SCIUI1_SPO0_SRVW4_SSO0_SVW4_SNLL0_TSGRA0_TSGRB0_TT1_96_TLDS1_UMLDSA0_UMLDSB1_USFGROn1_VAW2_VSn1_VW1_VWB1_VFLRP0_WSGRA1_WSGRB1_WS64_WG128_2_1_WGM1
│  └─ nan triton
│     ├─ nan <autotune>
│     │  └─ nan matmul_kernel
│     ├─ nan _ZN2at6native29vectorized_elementwise_kernelILi4ENS0_11FillFunctorIiEENS_6detail5ArrayIPcLi1EEEEEviT0_T1_
│     └─ nan matmul_kernel
├─ 1.624 matmul_1152_1152_1152
```

in that case the percentages of the ROOT value don't even add up to 100%.

Reply:
Sure, that's a good point.

Proton lacks a definition of "exclusive" metrics, but that requires more refactoring. I would say any real "inclusive" metrics can have a ratio. Currently, what we defined as "(inc)" are not real "inclusive" metrics.

For now, let's mark all these metrics as exclusive and do not allow derivation of percentage or ratio, including:

- avg_time_factor_dict
- derivable_metrics
- util

In contrast, other metrics and time_factor_dict can calculate a ratio metric.

Reply:
Updated